mount/mount_unix_test.go

```diff
@@ -251,3 +252,42 @@ func TestRecursiveUnmountTooGreedy(t *testing.T) {
 		t.Fatal("expected dir-other to be mounted, but it's not")
 	}
 }
+
+func TestRecursiveUnmount_SubMountFailsToUnmount(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("root required")
+	}
+
+	tmp, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+
+	// Create a bunch of tmpfs mounts. Make sure "dir" itself is not
+	// a mount point, or we'll hit the fast path in RecursiveUnmount.
+	dirs := []string{"dir-other", "dir/subdir1", "dir/subdir1/subsub", "dir/subdir2/subsub"}
+	for _, d := range dirs {
+		dir := path.Join(tmp, d)
+		if err := os.MkdirAll(dir, 0o700); err != nil {
+			t.Fatal(err)
+		}
+		if err := Mount("tmpfs", dir, "tmpfs", ""); err != nil {
+			t.Fatal(err)
+		}
+		//nolint:errcheck
+		defer Unmount(dir)
+	}
+
```

Comment:
I see that you copied the "TooGreedy" test except for its sanity check part. Do you think it's not necessary?